composer/cli/launcher.py

```diff
@@ -107,30 +120,87 @@ def parse_args():
     parser = get_parser()
 
     args = parser.parse_args()
-    if args.world_size == -1:
+
+    env_var_mapping = {
+        "nproc": "LOCAL_WORLD_SiZE",
+        "world_size": "WORLD_SIZE",
+        "base_rank": "BASE_RANK",
+        "node_rank": "NODE_RANK",
+        "master_addr": "MASTER_ADDR",
+        "master_port": "MASTER_PORT",
+    }
+
+    for arg_name, env_var_name in env_var_mapping.items():
+        if getattr(args, arg_name) is None:
+            if env_var_name in os.environ:
+                env_var_value = os.environ[env_var_name]
+                if env_var_name != "MASTER_ADDR":
+                    env_var_value = int(env_var_value)
+                setattr(args, arg_name, env_var_value)
+
+    if args.world_size is None:
         args.world_size = args.nproc
 
-    if args.node_rank == -1:
-        if args.base_rank % args.nproc != 0:
-            raise ValueError("node_rank not specified, but unable to infer since nodes appear to "
-                             "have different amounts of processes.")
-        args.node_rank = args.base_rank // args.nproc
+    if args.world_size < args.nproc:
+        raise ValueError(f"world_size({args.world_size}) cannot be less than nproc({args.nproc})")
 
-    return args
+    is_multinode = args.world_size > args.nproc
+
+    if is_multinode:
```

Comment:
Could this validation be moved to the `launch_procs` function below? If someone calls launch_procs functionally, then it would be nice if they could benefit from this validation.